sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala

```diff
@@ -104,12 +104,24 @@ abstract class FileWriteBuilder(options: DataSourceOptions)
       options: Map[String, String],
       dataSchema: StructType): OutputWriterFactory
 
+  /**
+   * Returns whether this format supports the given [[DataType]] in write path.
+   * By default all data types are supported.
+   */
+  def supportDataType(dataType: DataType): Boolean = true
```

Comment:
This name is awkward. Similar methods or traits use "supports" instead of "support". I think this should as well.

Reply:
I proposed to change the name as "supports..." in previous PR: https://github.com/apache/spark/pull/23639

See the opposed comment here:  https://github.com/apache/spark/pull/23639#pullrequestreview-196362643

Reply:
This one is different. We are in a new class, and there is no `supportXXX` method in this class that we need to follow.